roles/build-ansible-test-images/tasks/main.yml

```diff
@@ -1,31 +1,11 @@
 - name: Ensure dependencies are available
   include_tasks: pre-requirements.yml
 
-- when: images_build | bool
-  block:
-    - name: Create temporary directory
-      ansible.builtin.tempfile:
-        state: directory
-        suffix: .test
-      register: remote_tmp_dir
-
-    - name: Copy repository files to temporary directory
-      ansible.builtin.copy:
-        src: '{{ playbook_dir | dirname }}/ansible-test/'
-        dest: '{{ remote_tmp_dir.path }}/ansible-test/'
-        mode: preserve
-
-    - name: Build ansible-test container image
-      ansible.builtin.command:
-        cmd: "{{ remote_tmp_dir.path }}/{{ item.script }} {{ item.name }}:{{ item.tag }}"
-      loop: "{{ images_available }}"
-
-  always:
-    - name: Delete temporary directory
-      file:
-        path: "{{ remote_tmp_dir.path }}"
-        state: absent
-      no_log: yes
+- name: Build ansible-test container image
+  ansible.builtin.command:
+    cmd: "{{ ansible_user_dir }}/src/github.com/ansible-community/images/{{ item.script }} {{ item.name }}:{{ item.tag }}"
```

Comment:
Hmmm, on a second thought, this would work with Zuul but then breaks the ability to run it locally (unless that very specific path is used).

Zuul defines a "zuul" dictionary of variables so we can test if it's there and set the path accordingly, perhaps something like this ?
```yaml
- name: Set path to the repository
  vars:
    _with_zuul: "{{ ansible_user_dir }}/src/github.com/ansible-community/images"
    _without_zuul: "{{ playbook_dir | dirname }}"
  set_fact:
    _repository: "{{ (zuul is defined) | ternary(_with_zuul, _without_zuul) }}"
```

Reply:
In case zuul isn't around, the files still have to be copied over to the remote (except if remote happens to be equal to the controller).

Reply:
> In case zuul isn't around, the files still have to be copied over to the remote (except if remote happens to be equal to the controller).

Good point -- that's in fact how I have been testing it locally rather than on a remote host.
I would put ``localhost`` in the playbook's ``hosts`` (like antsibull) but it wouldn't work with Zuul since executing against localhost is very restricted.

Let's merge this as-is for now, it will also allow the post job to run and upload new image builds.
I'll open another PR for the approach I've suggested.